application/src/main/java/org/thingsboard/server/utils/TbNodeUpgradeUtils.java

```
@@ -16,27 +16,61 @@
 
+@Slf4j
 public class TbNodeUpgradeUtils {
 
-    public static void upgradeConfigurationAndVersion(RuleNode node, RuleNodeClassInfo nodeInfo) throws Exception {
+    public static void upgradeConfigurationAndVersion(RuleNode node, RuleNodeClassInfo nodeInfo) {
         JsonNode oldConfiguration = node.getConfiguration();
+        int configurationVersion = node.getConfigurationVersion();
+
+        int currentVersion = nodeInfo.getCurrentVersion();
+        var configClass = nodeInfo.getAnnotation().configClazz();
+
         if (oldConfiguration == null || !oldConfiguration.isObject()) {
-            var configClass = nodeInfo.getAnnotation().configClazz();
-            node.setConfiguration(JacksonUtil.valueToTree(configClass.getDeclaredConstructor().newInstance().defaultConfiguration()));
+            log.warn("Failed to upgrade rule node with id: {} type: {} fromVersion: {} toVersion: {}. " +
+                            "Current configuration is null or not a json object. " +
+                            "Going to set default configuration ... ",
+                    node.getId(), node.getType(), configurationVersion, currentVersion);
+            node.setConfiguration(getDefaultConfig(configClass));
         } else {
-            var tbVersionedNode = (TbNode) nodeInfo.getClazz().getDeclaredConstructor().newInstance();
-            TbPair<Boolean, JsonNode> upgradeResult = tbVersionedNode.upgrade(node.getConfigurationVersion(), oldConfiguration);
-            if (upgradeResult.getFirst()) {
-                node.setConfiguration(upgradeResult.getSecond());
+            var tbVersionedNode = getTbVersionedNode(nodeInfo);
+            try {
+                TbPair<Boolean, JsonNode> upgradeResult = tbVersionedNode.upgrade(configurationVersion, oldConfiguration);
+                if (upgradeResult.getFirst()) {
+                    node.setConfiguration(upgradeResult.getSecond());
+                }
+            } catch (TbNodeException e) {
```

catch all exceptions. not only TbNodeException